go-controller/pkg/node/default_node_network_controller.go

```diff
@@ -822,6 +822,15 @@ func (nc *DefaultNodeNetworkController) Start(ctx context.Context) error {
 		return fmt.Errorf("failed to set node zone annotation for node %s: %w", nc.name, err)
 	}
 
+	// NOTE: An assumption we make here is that `setupOVNNode(node)` has already ran above
+	// and updated the `config.Default.EncapIP` configuration.
+	if config.OvnKubeNode.Mode != types.NodeModeDPUHost {
+		klog.Infof("Setting EncapIp %s in node annotation", config.Default.EncapIP)
+		if err = util.SetNodeEncapIp(nodeAnnotator, config.Default.EncapIP); err != nil {
```

Comment:
I think that the encap ip for the local node is primarily managed from `node_ip_handler_linux.go`. you might want to take a look there to verify that is also doing what you expect it to do.

Reply:
@jcaamano Figured it out! Thanks for the help! It is OVN doing it because there is a discrepancy via OvS external ID (external_ids:ovn-encap-ip) and OVN SBDB. If I set `external_ids:ovn-encap-ip` in OvS to the other encap-ip then it would change in the sbdb's encap table.

Reply:
In conclusion, this is a non issue since there should not be any discrepancy between OvS and OVN SBDB for local nodes.